### PR TITLE
Add google deployment descriptors

### LINKED ISSUE #271: Add deployment descriptors for google

### PR DESCRIPTION
**Pull request by chrismatix**
fixes #271

### CODE REVIEW

File: infrastructure/airy.conf.tpl [hunk 1]
```diff
@@ -1,6 +1,8 @@
 FB_APP_ID=changeme
 FB_APP_SECRET=changeme
 FB_WEBHOOK_SECRET=changeme
+GOOGLE_PARTNER_KEY=changeme
+GOOGLE_SA_FILE={"type":"service_account","project_id":"airy","private_key_id":"no","private_key":"nokey","client_email":"no","client_id":"no","auth_uri":"no","token_uri":"no","no":"no","client_x509_cert_url":"no"}
```
Comment (chrismatix) [inline infrastructure/airy.conf.tpl:5]: The app validates the variable structure on startup, which is why we have to put this there until we do dynamic startups :)

File: infrastructure/deployments/api-admin.yaml [hunk 2]
```diff
@@ -33,8 +33,6 @@ spec:
           value: "<fb_app_id>"
         - name: FACEBOOK_APP_SECRET
           value: "<fb_app_secret>"
-        - name: SERVICE_NAME
```
Comment (chrismatix) [inline infrastructure/deployments/api-admin.yaml:36]: These are only used by users of the producer healthchecks (the webhooks)

File: infrastructure/deployments/sources-google-webhook.yaml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+                - name: Health-Check
+                  value: health-check
+            initialDelaySeconds: 60
+            periodSeconds: 10
+            failureThreshold: 3
+        - name: ngrok
+          command:
+            - /bin/bash
+            - -c
+            - /usr/local/bin/ngrok -config=/etc/ngrok/config.yml -log=stdout -subdomain="<ngrok_client_string>" 8080
```
Comment (ljupcovangelski) [inline infrastructure/deployments/sources-google-webhook.yaml:50]: This should be something like `<ngrok_google_client>` and the other one `<ngrok_facebook_client>`. They should be different and are provided at provisioning time

File: infrastructure/deployments/sources-google-webhook.yaml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  namespace: default
+data:
+  config.yml: |
+    server_addr: proxy.tunnel.airy.co:4443
+    trust_host_root_certs: true
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: public-urls
```
Comment (ljupcovangelski) [inline infrastructure/deployments/sources-google-webhook.yaml:78]: This already exists in `sources-facebook-webhook.yaml ` , where the FB_WEBHOOK_PUBLIC_URL is set.

I think would be better to have one configmap which holds all the public urls, outside of both deployment yamls.

Comment (chrismatix) [reply infrastructure/deployments/sources-google-webhook.yaml:78]: Ok, thanks. I will get right on it!

File: infrastructure/deployments/sources-google-webhook.yaml (hunk 3 above)
Comment (chrismatix) [reply infrastructure/deployments/sources-google-webhook.yaml:50]: Will do! Thank you!

File: infrastructure/deployments/sources-facebook-webhook.yaml [hunk 5]
```diff
@@ -33,13 +33,28 @@ spec:
...
+            initialDelaySeconds: 60
+            periodSeconds: 10
+            failureThreshold: 3
         - name: ngrok
           command:
             - /bin/bash
             - -c
-            - /usr/local/bin/ngrok -config=/etc/ngrok/config.yml -log=stdout -subdomain="<ngrok_client_string>" 8080
-          env:
+            - /usr/local/bin/ngrok -config=/etc/ngrok/config.yml -log=stdout -subdomain="$WEBHOOK_SUBDOMAIN" 8080
```
Comment (chrismatix) [inline infrastructure/deployments/sources-facebook-webhook.yaml:50]: @ljupcovangelski Is this ok now? :)

File: infrastructure/scripts/airy-conf.sh [hunk 6]
```diff
@@ -48,15 +50,20 @@ sed -i "s/<fb_app_secret>/${config[FB_APP_SECRET]}/" ~/airy-core/api-admin.yaml
...
 sed -i "s/<mail_password>/${config[MAIL_PASSWORD]}/" ~/airy-core/api-auth.yaml
 sed -i "s/<mail_from>/${config[MAIL_FROM]}/" ~/airy-core/api-auth.yaml
 
 # Generate random string for the ngrok webhook
-RANDOM_INGRESS_ID=`cat /dev/urandom | env LC_CTYPE=C tr -dc a-z0-9 | head -c 16; echo`
-sed -i "s/<ngrok_client_string>/fb-${RANDOM_INGRESS_ID}/" ~/airy-core/sources-facebook-webhook.yaml
+GOOGLE_INGRESS_ID=`cat /dev/urandom | env LC_CTYPE=C tr -dc a-z0-9 | head -c 16; echo`
+FACEBOOK_INGRESS_ID=`cat /dev/urandom | env LC_CTYPE=C tr -dc a-z0-9 | head -c 16; echo`
+sed -i "s/<ngrok_facebook_client>/fb-${FACEBOOK_INGRESS_ID}/" ~/airy-core/public-urls-config.yaml
+sed -i "s/<ngrok_google_client/gl-${GOOGLE_INGRESS_ID}/" ~/airy-core/public-urls-config.yaml
```
Comment (chrismatix) [inline infrastructure/scripts/airy-conf.sh:66]: Wanted to make this a one-liner, but sedly I was unable

**Review by pascal-airy — APPROVED** [review]
nice that you included some gardening 👍

File: infrastructure/scripts/airy-conf.sh (hunk 6 above)
Comment (pascal-airy) [reply infrastructure/scripts/airy-conf.sh:66]: oh wow..

Comment (lucapette) [reply infrastructure/scripts/airy-conf.sh:66]: would probably keep related lines together (like we do fb first and then google or the other way around)

**Review by lucapette — COMMENTED** [review]
We have to regenerate the create topics script

**Review by lucapette — DISMISSED** [review]
so  the script was already run!

File: infrastructure/deployments/sources-facebook-webhook.yaml (hunk 5 above)
Comment (ljupcovangelski) [reply infrastructure/deployments/sources-facebook-webhook.yaml:50]: Yeap :+1:

**Review by ljupcovangelski — APPROVED** [review]
Looks great :+1:

**Pull request merged**